repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/sql/QueryFolderPreProcessor.java

```diff
@@ -80,12 +91,191 @@ private void processCreate(AtlasStruct entity) throws AtlasBaseException {
+                                                          String folderQualifiedName) {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("updateChildAttributesOnUpdatingCollection");
+        Map<String, Object> updatedAttributes = new HashMap<>();
+        String qualifiedName            =   childVertex.getProperty(QUALIFIED_NAME, String.class);
+        String updatedQualifiedName     =   qualifiedName.replaceAll(currentCollectionQualifiedName, newCollectionQualifiedName);
+
+        if (!qualifiedName.equals(updatedQualifiedName)) {
+            AtlasGraphUtilsV2.setEncodedProperty(childVertex, QUALIFIED_NAME, updatedQualifiedName);
+            updatedAttributes.put(QUALIFIED_NAME, updatedQualifiedName);
+        }
+
+        if(!currentCollectionQualifiedName.equals(newCollectionQualifiedName)) {
+            AtlasGraphUtilsV2.setEncodedProperty(childVertex, COLLECTION_QUALIFIED_NAME, newCollectionQualifiedName);
+            updatedAttributes.put(COLLECTION_QUALIFIED_NAME, newCollectionQualifiedName);
+        }
+
+        AtlasGraphUtilsV2.setEncodedProperty(childVertex, PARENT_QUALIFIED_NAME, folderQualifiedName);
+
+        //update system properties
+        GraphHelper.setModifiedByAsString(childVertex, RequestContext.get().getUser());
+        GraphHelper.setModifiedTime(childVertex, System.currentTimeMillis());
+
+        updatedAttributes.put(PARENT_QUALIFIED_NAME, folderQualifiedName);
+
+        //Record the updated child entities, it will be used to send notification and store audit logs
+        recordUpdatedChildEntities(childVertex, updatedAttributes);
+
+        RequestContext.get().endMetricRecord(metricRecorder);
+    }
+
+    /**
+     * Record the updated child entities, it will be used to send notification and store audit logs
+     * @param entityVertex Child entity vertex
+     * @param updatedAttributes Updated attributes while updating required attributes on updating collection
+     */
+    private void recordUpdatedChildEntities(AtlasVertex entityVertex, Map<String, Object> updatedAttributes) {
+        RequestContext requestContext = RequestContext.get();
+        AtlasPerfMetrics.MetricRecorder metricRecorder = requestContext.startMetricRecord("recordUpdatedChildEntities");
+        AtlasEntity entity = new AtlasEntity();
+        entity = entityRetriever.mapSystemAttributes(entityVertex, entity);
+        entity.setAttributes(updatedAttributes);
+        requestContext.cacheDifferentialEntity(new AtlasEntity(entity));
+
+        AtlasEntityType entityType = typeRegistry.getEntityTypeByName(entity.getTypeName());
+
+        //Add the min info attributes to entity header to be sent as part of notification
+        if(entityType != null) {
+            AtlasEntity finalEntity = entity;
+            entityType.getMinInfoAttributes().values().stream().filter(attribute -> !updatedAttributes.containsKey(attribute.getName())).forEach(attribute -> {
```

Comment:
can we use entityType.getMinInfoAttributes().keySet() to filter attributes here?

Reply:
Kept this to use this function `entityRetriever.getVertexAttribute(entityVertex, attribute);` which is more sophisticated to fetch attributes. That need whole `attribute` there.